stl/src/alias.asm

```diff
@@ -6,19 +6,51 @@
 ; Note that while this is a masm file, it contains no code
 ; masm does generate a .text section with zero size
 
-IFDEF RAX ; note: this only works with ifdef, not ifndef
-ELSE
-.MODEL FLAT, STDCALL
-ENDIF
+; mangle identifiers for stdcall calling convention
+; on x86_64 this does nothing, on x86 it adds a leading
+; underscore and @sz, sz should be the size of the parameters
+mangle MACRO name, sz
+    ; This checks if we're in x86_64 mode (using ml64)
+    ; note that RAX isn't really an assembler symbol,
+    ; and this check only works in the IFDEF direction
+    ; That is:
+    ; IFDEF RAX
+    ;   meow
+    ; ENDIF
+    ; will expand to meow on only x64 system, however:
+    ; IFNDEF RAX
+    ;   meow
+    ; ENDIF
+    ; will expand to meow on _BOTH_ x64 and x86 systems.
+    IFDEF RAX
+        EXITM <name>
+    ELSE
+        EXITM <_&name&@&sz&>
+    ENDIF
+ENDM
 
-EXTERN __imp_InitOnceBeginInitialize :PROTO :PTR PTR VOID, :DWORD, :PTR DWORD, :PTR PTR VOID
-EXTERN __imp_InitOnceComplete :PROTO :PTR PTR VOID, :DWORD, :PTR VOID
+; generate the name of the imp symbol for a function
+imp_name MACRO name, sz
+    EXITM @CatStr(__imp_, mangle(name, sz))
+ENDM
+
+create_alias MACRO oldname, newname, size
+    ; use EXTERN name : PROC instead of
+    ; EXTERN name : PROTO ...
+    ; to avoid masm adding underscores to the names
+    ; of (object) symbols.
+    ; Note that the MASM symbol always has the name that appears in
+    ; the source file (after macro expansion), but when _using_ the symbol
+    ; masm will add underscores
+    EXTERN imp_name(newname, size) : PROC
+    ; the documentation for "ALIAS <alias> = <actual-name>" says angle brackets
+    ; are required around the alias, but this is not the case, it seems to just require
+    ; a single "text item", which angle brackets create, but which is also the result
+    ; of expanding these macros
+    ALIAS imp_name(oldname, size)=imp_name(newname,size)

```

Comment:
Assuming the spaces are insignificant, can we:
```suggestion
    ALIAS imp_name(oldname, size) = imp_name(newname, size)
```
for consistency with our normal conventions for spaces around operators?